File: src/main/java/xd/marcin/hotel/entities/Reservation.java
package xd.marcin.hotel.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import java.time.ZonedDateTime;
import java.util.Date;


@Entity
@Data
@NoArgsConstructor
//@AllArgsConstructor
public class Reservation {
    @Id @GeneratedValue
    @Column
    private Integer id;

    @Column
    private Integer beds;

    @Column
    private String comments;

    @ManyToOne
    private Client client;

    @Column
    private Date fromDate;

    @Column
    private Date toDate;
}

File: frontend/src/app/models/reservation.ts
export class Reservation {
    // private id: number;
    // private beds: number;
    // private comments: string;
    // private client: any[];
    // private fromDate: string;
    // private toDate: string;
    id: number;
    beds: number;
    comments: string;
    client: any[];
    fromDate: string;
    toDate: string;

    constructor(obj?: any) {
        this.id = (obj && obj.id) || null;
        this.beds = (obj && obj.id) || null;
        this.comments = (obj && obj.comments) || null;
        this.client = (obj && obj.client) || null;
        this.fromDate = (obj && obj.fromDate) || null;
        this.toDate = (obj && obj.toDate) || null;
  }

  // getId(): number {
  //   return this.id;
  // }
  // getBeds(): number {
  //   return this.beds;
  // }

  // isFinished(): boolean {
  //   return this.finished;
  // }
  // toggleStatus(): void {
  //   this.finished = !this.finished;
  // }
}
